### PR TITLE
[DO NOT MERGE] Propogate dataset (data-* attributes) from amp-img to img

### PR DESCRIPTION
**Pull request by erwinmombay**
fixes https://github.com/ampproject/amphtml/issues/24787

Will not merge this as this review will only be used for the code jam

### CODE REVIEW

**Comment by erwinmombay** [comment]
@cathyxz @nainar please review :)

**Comment by erwinmombay** [comment]
should we re-propagate on mutation callback?

File: builtins/amp-img.js [hunk 1]
```diff
@@ -178,13 +178,24 @@ export class AmpImg extends BaseElement {
...
 
     this.element.appendChild(this.img_);
   }
 
+  /**
+   * Propagate the dataset key/value pairs in the `amp-img` element to the
+   * backing `img` element.
+   */
+  propagateDataset_() {
+    for (const key in this.element.dataset) {
```
Comment (jridgewell) [inline builtins/amp-img.js]: This can get out of sync very easily if data attribute is removed from the amp-img.

Comment (jridgewell) [reply builtins/amp-img.js]: You just have to listen in `mutatedAttributesCallback`.

Comment (erwinmombay) [reply builtins/amp-img.js]: done. do you mind taking another look? thanks. i just delete and reapply everything again. i wasn't sure if there was anything more granular like a delete entry on the object that is passed into `mutatedAttributesCallback`

**Comment by erwinmombay** [comment]
@choumx do you mind taking a look at this too? thanks

File: builtins/amp-img.js [hunk 2]
```diff
@@ -178,13 +179,27 @@ export class AmpImg extends BaseElement {
...
 
     this.element.appendChild(this.img_);
   }
 
+  /**
+   * Propagate the dataset key/value pairs in the `amp-img` element to the
+   * backing `img` element.
+   */
+  propagateDataset_() {
+    for (const key in this.img_.dataset) {
```
Comment (jridgewell) [inline builtins/amp-img.js]: Nits: Save both datasets into local variables, repeated accesses from `el.dataset` are slow.

Comment (erwinmombay) [reply builtins/amp-img.js]: done.

**Pull request closed without merging**